### ISSUE TITLE
[Feature Request] Snooze chats, channels, or groups

### ISSUE THREAD

**Issue by jmorgannz** [kotatogram/kotatogram-desktop#407]
### Is your feature request related to a problem?

Sometimes you want to completely ignore a person, a channel, or a group for a period of time.  
You don't want to lose the messages that come in during that period of time, but you also don't want notifications to come in - even with an unread counter.  

For people who need to manage stress, being able to 'disconnect' for a period of time to be in a space of one's own control is crucial.  
Seeing an unread notification is enough to break that if one is in a stressed state - because the stress of knowing someone is trying to contact you and waiting adds to the situation.  

Likewise, blocking contacts who are likely to raise stress and then unblocking them later means that messages sent whilst blocked are lost.  

It is not even possible to block a group or channel.  

### Describe the solution you'd like

A 'Snooze' function with similar UI to Mute and its various options - that being either a set timeframe or until manually unmuted.  

This should be able to be applied to any message source - whether an individual contact, group, or channel.  

When snoozed, the source sending you messages would not have any idea they are snoozed - their messages would simply be left on delivered, but unread.  

When snoozed, no messages, notifications, or unread counts would be shown for the snoozed source.  
These would be postponed until the source reaches the end of its snooze period or is manually unsnoozed.  

### Describe alternatives you've considered

Closing all instances of Telegram on all devices is the only alternative - but it does not allow fine grained control of which sources are snoozed and which are allowed through.  
It is crucial to be able to have control of who is and is not snoozed so one can remain contactable but manage their exposure to outside disruption.  

Another alternative feature could be called 'trusted mode' or something, where you have certain trusted contacts and everyone else is untrusted.  
Then one could place Telegram into 'trusted only' mode, and only trusted contacts would be allowed to correspond, whilst to all other contacts and groups the user would appear offline until trusted only mode is exited.

**Comment by EricKotato** [comment]
Here's another alternative: you can make a folder with all chats that you want to see during this period of time, and set this folder as default one.

But I'll think about implementing something like this.

**Comment by jmorgannz** [comment]
Hi,

Yeah this doesn't really work as someone can still bomb you on other
folders and you will see the unread counts.
I have written an app with the API which quickly reads messages from
certain groups and peers during a set period (DND period), or I can
manually set FORCE DND on or off.
This means that the unread count never shows (they are also muted so no
message sound plays)

When DND mode exits, the app remembers which peers had sent messages during
DND, and re-marks the dialogs as unread.

But still it would be great if this was part of telegram itself.
